aptos-move/mvhashmap/src/versioned_code.rs

```diff
@@ -32,30 +37,27 @@ struct Entry<V: TransactionWrite> {
 
 /// A VersionedValue internally contains a BTreeMap from indices of transactions
 /// that update the given access path alongside the corresponding entries.
-struct VersionedValue<V: TransactionWrite, X: Executable> {
+struct VersionedValue<V: TransactionWrite + Send + Sync, X: Executable> {
     versioned_map: BTreeMap<TxnIndex, CachePadded<Entry<V>>>,
 
-    /// Executable based on the storage version of the module.
-    base_executable: Option<Arc<X>>,
     /// Executables corresponding to published versions of the module, based on hash.
-    executables: HashMap<HashValue, Arc<X>>,
+    executables: HashMap<HashValue, X>,
 }
 
 /// Maps each key (access path) to an interal VersionedValue.
-pub struct VersionedCode<K, V: TransactionWrite, X: Executable> {
+pub struct VersionedCode<
+    K: ModulePath + Hash + Clone + Eq + Send + Sync,
+    V: TransactionWrite + Send + Sync,
+    X: Executable,
+> {
     values: DashMap<K, VersionedValue<V, X>>,
+    base_executables: Arc<ExecutableStore<K, X>>,
```

Comment:
I'd prefer to split this cache to a separate change but anyway

Reply:
Let me do it, I really hate large PRs and want to follow proper practices even for habits.